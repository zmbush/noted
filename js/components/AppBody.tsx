// Copyright 2021 Zachary Bush.
//
// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
// option. This file may not be copied, modified, or distributed
// except according to those terms.
//
import Mousetrap from 'mousetrap';

import * as React from 'react';
import { Route, Routes } from 'react-router-dom';

import { Grid } from '@mui/material';

import FilteredNoteList from 'components/FilteredNoteList';
import Note from 'components/Note';
import NoteList from 'components/NoteList';
import { NoteWithTags } from 'data/types';

type NewNoteProps = {
  newNote: boolean;
  search: string;
  onNewNoteCancel: () => void;
};

export const NewNote = ({ newNote, search, onNewNoteCancel }: NewNoteProps) => {
  if (!newNote) {
    return null;
  }
  return (
    <Grid item xs={12}>
      <Note
        depth={1}
        search={search}
        note={{
          title: search,
          body: '',
        }}
        onNewNoteCancel={onNewNoteCancel}
      />
    </Grid>
  );
};

type Props = {
  notes: Map<number, NoteWithTags>;
  createNewShortcut: (
    e: Mousetrap.ExtendedKeyboardEvent | React.SyntheticEvent,
    combo?: string,
  ) => void;
} & NewNoteProps;

const AppBody = ({ notes, createNewShortcut, newNote, search, onNewNoteCancel }: Props) => {
  const filteredNoteList = <FilteredNoteList depth={1} search={search} />;

  const mainNoteList = (
    <NoteList
      createFromSearch={createNewShortcut}
      parent_note_id={null}
      depth={1}
      notes={notes}
      search={search}
    />
  );

  return (
    <Grid
      container
      spacing={1}
      sx={{
        '@media print': {
          marginTop: 0,
          display: 'block',
        },
        marginTop: '75px',
      }}
    >
      <NewNote newNote={newNote} search={search} onNewNoteCancel={onNewNoteCancel} />
      <Routes>
        <Route path='/' element={mainNoteList} />
        <Route path='/archive' element={mainNoteList} />
        <Route path='/note/:ids' element={filteredNoteList} />
        <Route path='/disambiguation/:ids' element={filteredNoteList} />
      </Routes>
    </Grid>
  );
};

export default AppBody;
