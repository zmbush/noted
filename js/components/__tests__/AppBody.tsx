// Copyright 2021 Zachary Bush.
//
// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
// option. This file may not be copied, modified, or distributed
// except according to those terms.
//
import { shallow } from 'enzyme';

import * as React from 'react';

import AppBody, { NewNote } from '../AppBody';

describe('<AppBody />', () => {
  test('matches snapshot', () => {
    const wrapper = shallow(
      <AppBody
        createNewShortcut={() => {}}
        notes={{}}
        newNote={false}
        search=''
        onNewNoteCancel={() => {}}
      />,
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(NewNote)).toMatchSnapshot();

    wrapper.setProps({
      search: 'Search Term',
      newNote: true,
    });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(NewNote)).toMatchSnapshot();
  });
});

describe('<NewNote />', () => {
  test('matches snapshot', () => {
    const wrapper = shallow(<NewNote newNote={false} search='' onNewNoteCancel={() => {}} />);
    expect(wrapper).toMatchSnapshot();

    wrapper.setProps({
      search: 'Search Term',
      newNote: true,
    });
    expect(wrapper).toMatchSnapshot();
  });
});
