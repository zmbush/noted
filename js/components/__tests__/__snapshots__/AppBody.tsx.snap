// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppBody /> matches snapshot 1`] = `
<ForwardRef(Grid)
  container={true}
  spacing={2}
  sx={
    Object {
      "@media print": Object {
        "display": "block",
        "marginTop": 0,
      },
      "marginTop": "75px",
    }
  }
>
  <NewNote
    newNote={false}
    onDeleteNote={[Function]}
    onUpdateNote={[Function]}
    search=""
  />
  <Routes>
    <Route
      element={
        <Memo(Connect(NoteList))
          createFromSearch={[Function]}
          depth={1}
          notes={Map {}}
          onDeleteNote={[Function]}
          onUpdateNote={[Function]}
          parent_note_id={null}
          search=""
        />
      }
      path="/"
    />
    <Route
      element={
        <Memo(Connect(FilteredNoteList))
          depth={1}
          onDeleteNote={[Function]}
          onUpdateNote={[Function]}
          search=""
        />
      }
      path="/note/:ids"
    />
    <Route
      element={
        <Memo(Connect(FilteredNoteList))
          depth={1}
          onDeleteNote={[Function]}
          onUpdateNote={[Function]}
          search=""
        />
      }
      path="/disambiguation/:ids"
    />
  </Routes>
</ForwardRef(Grid)>
`;

exports[`<AppBody /> matches snapshot 2`] = `
<NewNote
  newNote={false}
  onDeleteNote={[Function]}
  onUpdateNote={[Function]}
  search=""
/>
`;

exports[`<AppBody /> matches snapshot 3`] = `
<ForwardRef(Grid)
  container={true}
  spacing={2}
  sx={
    Object {
      "@media print": Object {
        "display": "block",
        "marginTop": 0,
      },
      "marginTop": "75px",
    }
  }
>
  <NewNote
    newNote={true}
    onDeleteNote={[Function]}
    onUpdateNote={[Function]}
    search="Search Term"
  />
  <Routes>
    <Route
      element={
        <Memo(Connect(NoteList))
          createFromSearch={[Function]}
          depth={1}
          notes={Map {}}
          onDeleteNote={[Function]}
          onUpdateNote={[Function]}
          parent_note_id={null}
          search="Search Term"
        />
      }
      path="/"
    />
    <Route
      element={
        <Memo(Connect(FilteredNoteList))
          depth={1}
          onDeleteNote={[Function]}
          onUpdateNote={[Function]}
          search="Search Term"
        />
      }
      path="/note/:ids"
    />
    <Route
      element={
        <Memo(Connect(FilteredNoteList))
          depth={1}
          onDeleteNote={[Function]}
          onUpdateNote={[Function]}
          search="Search Term"
        />
      }
      path="/disambiguation/:ids"
    />
  </Routes>
</ForwardRef(Grid)>
`;

exports[`<AppBody /> matches snapshot 4`] = `
<NewNote
  newNote={true}
  onDeleteNote={[Function]}
  onUpdateNote={[Function]}
  search="Search Term"
/>
`;

exports[`<NewNote /> matches snapshot 1`] = `""`;

exports[`<NewNote /> matches snapshot 2`] = `
<ForwardRef(Grid)
  item={true}
  xs={12}
>
  <Connect(Note)
    depth={1}
    new={true}
    note={
      Object {
        "body": "",
        "created_at": "",
        "id": -1,
        "tags": Array [],
        "title": "Search Term",
        "updated_at": "",
        "user_id": 0,
      }
    }
    onDeleteNote={[Function]}
    onUpdateNote={[Function]}
    search="Search Term"
  />
</ForwardRef(Grid)>
`;
